game/theater/theatergroundobject.py

```
@@ -44,56 +51,148 @@
 }
 
 
-GroupT = TypeVar("GroupT", ShipGroup, VehicleGroup)
+@dataclass
+class GroundUnit:
+    # Units can be everything.. Static, Vehicle, Ship.
+    id: int
+    name: str
+    type: str  # dcs.UnitType as string
+    position: PointWithHeading
+    ground_object: TheaterGroundObject
+    alive: bool = True
+    _unit_type: Optional[UnitType[Any]] = None
+
+    @staticmethod
+    def from_template(
+        id: int, unit_type: str, t: UnitTemplate, go: TheaterGroundObject
+    ) -> GroundUnit:
+        return GroundUnit(
+            id,
+            t.name,
+            unit_type,
+            PointWithHeading.from_point(t.position, Heading.from_degrees(t.heading)),
+            go,
+        )
+
+    @property
+    def unit_type(self) -> Optional[UnitType[Any]]:
+        if not self._unit_type:
+            try:
+                unit_type: Optional[UnitType[Any]] = None
+                dcs_type = db.unit_type_from_name(self.type)
+                if dcs_type and issubclass(dcs_type, VehicleType):
+                    unit_type = next(GroundUnitType.for_dcs_type(dcs_type))
+                elif dcs_type and issubclass(dcs_type, ShipType):
+                    unit_type = next(ShipUnitType.for_dcs_type(dcs_type))
+                self._unit_type = unit_type
+            except StopIteration:
+                logging.error(f"No UnitType for {self.type}")
```

This is a severe error, isn't it? Should raise an exception and the return type should be non-Optional.